src/bun.js/bindings/JSBunRequest.cpp

```diff
@@ -66,6 +70,39 @@ JSObject* JSBunRequest::cookies() const
     return nullptr;
 }
 
+extern "C" void* Request__clone(void* internalZigRequestPointer, JSGlobalObject* globalObject);
+
+JSBunRequest* JSBunRequest::clone(JSC::VM& vm, JSGlobalObject* globalObject)
+{
+    auto* structure = createJSBunRequestStructure(vm, defaultGlobalObject(globalObject));
+    auto* clone = this->create(vm, structure, Request__clone(this->wrapped(), globalObject), nullptr);
+
+    // Cookies and params are deep copied as they can be changed between the clone and original
+    if (auto* params = this->params()) {
+        auto* prototype = defaultGlobalObject(globalObject)->m_JSBunRequestParamsPrototype.get(globalObject);
+        auto* paramsClone = JSC::constructEmptyObject(globalObject, prototype);
+
+        auto propertyNames = PropertyNameArray(vm, JSC::PropertyNameMode::Strings, JSC::PrivateSymbolMode::Exclude);
```

Comment:
This would ideally use https://github.com/oven-sh/WebKit/blob/c5e9b9e327194f520af2c28679adb0ea1fa902ad/Source/JavaScriptCore/runtime/JSGlobalObjectFunctions.cpp#L1018-L1099 but they haven't exposed it as a method yet